src/jukebox/components/rfid/hardware/generic_nfcpy/generic_nfcpy.py

```diff
@@ -0,0 +1,90 @@
+# Standard imports from python packages
+import logging
+
+import nfc
+from nfc.clf import RemoteTarget
+
+# Import the ReaderBaseClass for common API. Leave as this line as it is!
+from components.rfid import ReaderBaseClass
+import jukebox.cfghandler
+
+# Also import the description string into this module, to make everything available in a single module w/o code duplication
+# Leave this line as is!
+from .description import DESCRIPTION
+
+# Create logger.
+# Logging is fully setup. Just replace '.new' with something meaningful and short
+logger = logging.getLogger('jb.rfid.new')
+# Get the global handler to the RFID config
+cfg = jukebox.cfghandler.get_handler('rfid')
+
+
+
+
+def query_customization() -> dict:
+    """
+    Query the user for reader parameter customization
+    """
+    return {}
+
+
+class ReaderClass(ReaderBaseClass):
+    """
+    The reader class for nfcpy supported NFC card readers.
+    """
+    def __init__(self, reader_cfg_key):
+        # Create a per-instance logger, just in case the reader will run multiple times in various threads
+        # Replace '.new' with something meaningful and short
+        self._logger = logging.getLogger(f'jb.rfid.new({reader_cfg_key})')
+        # Initialize the super-class. Don't change anything here
+        super().__init__(reader_cfg_key=reader_cfg_key, description=DESCRIPTION, logger=self._logger)
+
+        # Get the configuration from the rfid.yaml:
+        # Lock config around the access
+        #with cfg:
+        #    # Get a reference to the actual reader-specific config
```

Comment:
Is the driver parameter necessary here?

Reply:
nfcpy handles all supported devices automatically. The driver is only needed during the installation to blacklist the kernel driver. As there is a bash installation script I think I can just blacklist all of the possible ones from there. Then the code is completely driver agnostic.